features/HistoricalRequest.feature.md

```diff
@@ -0,0 +1,77 @@
+# Historical data request
+
+> A client can request the history for the device.
+
+## Background
+
+Given I have the fingerprint for a `PCA20035+solar` device in `fingerprint`
+
+And I store
+`$toMillis($fromMillis($millis(), '[Y9999]-[M99]-[D99]T[H99]:[m]:00Z'))` into
+`ts`
+
+## Scenario Outline: Device publishes the messages to nRF Cloud
+
+Given I store `ts - ${deductMsFromTS}` into `${storeName}`
+
+Then the device `${fingerprint:deviceId}` publishes the message with properties
+`${appId}`, `${messageType}`, `${data}`, and `${storeName}` to the topic
+`m/d/${fingerprint:deviceId}/d2c`
+
+### Examples
+
+| appId | messageType | data    | deductMsFromTS | storeName |
+| ----- | ----------- | ------- | -------------- | --------- |
+| SOLAR | DATA        | 3.40141 | 0              | ts1       |
+| SOLAR | DATA        | 3.75718 | 30000          | ts2       |
+| SOLAR | DATA        | 3.73368 | 60000          | ts3       |
+| SOLAR | DATA        | 3.58041 | 90000          | ts4       |
+| SOLAR | DATA        | 3.24925 | 120000         | ts5       |
```

Comment:
Should follow established format of devices publishing messages:

```suggestion
Given I store `ts - ${deductMsFromTS}` into `pastTs`

And the device `${fingerprint:deviceId}` publishes this message to the topic
`m/d/${fingerprint:deviceId}/d2c`

```json
{
  "appId": "${appId}",
  "messageType": "DATA",
  "ts": ${pastTs},
  "data": "${data}"
}
```

### Examples

| appId | messageType | data    | deductMsFromTS |
| ----- | ----------- | ------- | -------------- |
| SOLAR | DATA        | 3.40141 | 0              |
| SOLAR | DATA        | 3.75718 | 30000          |
| SOLAR | DATA        | 3.73368 | 60000          |
| SOLAR | DATA        | 3.58041 | 90000          |
| SOLAR | DATA        | 3.24925 | 120000         |

```